src/api/EventsAPI.ts

```diff
@@ -19,30 +16,101 @@ export type EventObject = {
   pointValue: number;
   requiresStaff: boolean;
   staffPointBonus: number;
+  facebookUrl?: string;
 };
 
-const handleErrors = (response: Response): Promise<EventsResponse> => {
+export type EventsArray = EventObject[];
+
+export type EventsResponse = {
+  error: unknown;
+  events: EventsArray;
+};
+
+export type EventResponse = {
+  error: unknown;
+  event: EventObject;
+};
+
+const handleErrors = (response: Response) => {
   if (!response.ok) {
     throw Error(response.statusText);
   }
   return response.json();
 };
 
-const getAllEvents = async (): Promise<EventsArray | undefined> => {
-  let apiurl = "https://api.acmucsd.com/api/v2/event/future";
-
-  // TODO: Fix test data for development purposes (images are not showing up)
-  // if (!process.env.NODE_ENV || process.env.NODE_ENV === "development") {
-  //   apiurl = "https://testing.api.acmucsd.com/api/v2/event/future"
-  // }
+const getAllEvents = async (
+  type: "past" | "future" | "" = ""
+): Promise<EventsArray | undefined> => {
+  const api_url = `${EVENT_API}/${type}`;
 
   try {
-    const response: Response = await fetch(apiurl);
+    const response: Response = await fetch(api_url);
     const result: EventsResponse = await handleErrors(response);
     return result.events;
   } catch (error) {
     return undefined;
   }
 };
 
-export { getAllEvents };
+const getEvent = async (uuid: string): Promise<EventObject | undefined> => {
+  let api_url = `${EVENT_API}/${uuid}`;
+
+  try {
+    const response: any = await fetch(api_url);
+    const result: EventResponse = await handleErrors(response);
+    return result.event;
+  } catch (error) {
+    return undefined;
+  }
+};
+
+const createDownloadFile = (textContent: string, title: string): void => {
+  const element = document.createElement("a");
```

Comment:
credit the stackoverflow where we lifted this from so we can maintain it better in the future